### ISSUE TITLE
Test fails when static libs are disabled

### ISSUE THREAD

**Issue by krisk0** [wbhart/flint2#107]
I have FLINT2.4.4 installed, and configure/compile/test ver  d7b9299a2399c7fe4f0e50e59ca6926ccac79f9a dated 1 oct.

With static libs turned on, configure/make/make check succeeds. However, turning off static results in compilation failure:

```
x86_64-pc-linux-gnu-gcc -O2 -march=native -pipe -DFLINT_CPIMPORT=\"/usr/share/flint/CPimport.txt\" -I/tmp/portage/sci-mathematics/flint-20141001/work/flint2-d7b9299a2399c7fe4f0e50e59ca6926ccac79f9a -I/usr/include -I/usr/include -I/usr/include test/t-powmod_ui_preinv.c ../build/ulong_extras/../../test_helpers.o -o ../build/ulong_extras/test/t-powmod_ui_preinv -L/tmp/portage/sci-mathematics/flint-20141001/work/flint2-d7b9299a2399c7fe4f0e50e59ca6926ccac79f9a -L/usr/lib -L/usr/lib -L/usr/lib -lflint -lmpfr -lgmp -lm -lntl -lpthread  -MMD -MP -MF ../build/ulong_extras/test/t-powmod_ui_preinv.d -MT "../build/ulong_extras/test/t-powmod_ui_preinv" -MT "../build/ulong_extras/test/t-powmod_ui_preinv.d"

/tmp/portage/sci-mathematics/flint-20141001/temp/ccW74vWX.o: In function `main':
t-powmod_ui_preinv.c:(.text.startup+0x107): undefined reference to `n_powmod_ui_preinv'
collect2: error: ld returned 1 exit status
make[1]: *** [../build/ulong_extras/test/t-powmod_ui_preinv] Error 1
```

This error occurs because subroutine n_powmod_ui_preinv() does not exist in flint-2.4.4 but test suite links to installed flint (-L/usr/lib -lflint) rather than freshly compiled.

Summing up: when static libs are off, test suite actually tests installed version of flint found in /usr/lib which is incorrect.

**Comment by wbhart** [comment]
I don't know how to fix this issue. The installed shared library is going to be used by the linker, if it exists. I don't see how to prevent it looking there.

Anyway, I can confirm that without a system flint in /usr/lib the tests pass with --disable-static.

It also works with --disable-shared.

**Comment by isuruf** [comment]
This is fixed in trunk with adding a rpath to the build directory.

**Comment by wbhart** [comment]
Do you mean it has been fixed, the user can fix it for themselves by specifying rpath or it can be fixed by us this way?

**Comment by isuruf** [comment]
It has been fixed in trunk.

**Comment by wbhart** [comment]
Thanks.
